language/tools/move-mv-llvm-compiler/src/stackless/llvm.rs

```diff
@@ -0,0 +1,359 @@
+    }
+
+    pub fn int8_type(&self) -> Type {
+        unsafe { Type(LLVMInt8TypeInContext(self.0)) }
+    }
+
+    pub fn int64_type(&self) -> Type {
+        unsafe { Type(LLVMInt64TypeInContext(self.0)) }
+    }
+}
+
+pub struct Module(LLVMModuleRef);
+
+impl Drop for Module {
+    fn drop(&mut self) {
+        unsafe {
+            LLVMDisposeModule(self.0);
+        }
+    }
+}
+
+impl AsMut<llvm_sys::LLVMModule> for Module {
+    fn as_mut(&mut self) -> &mut llvm_sys::LLVMModule {
+        unsafe { &mut *self.0 }
+    }
+}
+
+impl Module {
+    pub fn set_target(&self, triple: &str) {
+        unsafe {
+            LLVMSetTarget(self.0, triple.cstr());
+        }
+    }
+
+    pub fn set_source_file_name(&self, name: &str) {
+        unsafe { LLVMSetSourceFileName(self.0, name.as_ptr() as *const libc::c_char, name.len()) }
+    }
+
+    pub fn add_function(&self, name: &str, ty: FunctionType) -> Function {
+        unsafe { Function(LLVMAddFunction(self.0, name.cstr(), ty.0)) }
+    }
+
+    pub fn add_function_with_attrs(
+        &self,
+        name: &str,
+        ty: FunctionType,
+        attrs: &[AttributeKind],
+    ) -> Function {
+        unsafe {
+            let cx = LLVMGetModuleContext(self.0);
+            let attrs = attrs
+                .iter()
+                .map(|a| LLVMRustCreateAttrNoValue(cx, *a))
+                .collect::<Vec<_>>();
+
+            let llfn = self.add_function(name, ty);
+
+            AddFunctionAttributes(llfn.0, AttributePlace::Function, &attrs);
+
+            llfn
+        }
+    }
+
+    pub fn get_named_function(&self, name: &str) -> Option<Function> {
+        unsafe {
+            let llfn = LLVMGetNamedFunction(self.0, name.cstr());
+            if !llfn.is_null() {
+                Some(Function(llfn))
+            } else {
+                None
+            }
+        }
+    }
+
+    pub fn verify(&self) {
+        use llvm_sys::analysis::*;
+        unsafe {
+            LLVMVerifyModule(
+                self.0,
+                LLVMVerifierFailureAction::LLVMAbortProcessAction,
+                ptr::null_mut(),
+            );
+        }
+    }
+}
+
+pub struct Builder(LLVMBuilderRef);
+
+impl Drop for Builder {
+    fn drop(&mut self) {
+        unsafe {
+            LLVMDisposeBuilder(self.0);
+        }
+    }
+}
+
+impl Builder {
+    pub fn position_at_end(&self, bb: BasicBlock) {
+        unsafe {
+            LLVMPositionBuilderAtEnd(self.0, bb.0);
+        }
+    }
+
+    pub fn build_alloca(&self, ty: Type, name: Name) -> Alloca {
+        unsafe { Alloca(LLVMBuildAlloca(self.0, ty.0, name.0.cstr())) }
+    }
+
+    pub fn store_param_to_alloca(&self, param: Parameter, alloca: Alloca) {
+        unsafe {
+            LLVMBuildStore(self.0, param.0, alloca.0);
+        }
+    }
+
+    pub fn load_store(&self, ty: Type, src: Alloca, dst: Alloca) {
+        unsafe {
+            let tmp_name = Name::kind("load_store_tmp");
```

Comment:
Do you thing we should establish a naming convention for consistency?

Reply:
Probably, though I am not sure what yet. In the last patch I removed this `Name` abstraction because I thought it was overkill and not accomplishing much.